pymc3/ode/ode.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import scipy
+import theano
+import theano.tensor as tt
+from pymc3.ode.utils import augment_system, ODEGradop
+THEANO_FLAG = 'compute_test_value=ignore'
+
+class DifferentialEquation(theano.Op):
+
+    '''
+    Specify an ordinary differential equation
+
+    .. math::
+        \dfrac{dy}{dt} = f(y,t,p) \quad y(t_0) = y_0
+
+    Parameters
+    ----------
+
+    func : callable
+        Function specifying the differential equation
+    t0 : float
+        Time corresponding to the initial condition
+    times : array
+        Array of times at which to evaluate the solution of the differential equation.
+    n_states : int
+        Dimension of the differential equation.  For scalar differential equations, n_states=1.
+        For vector valued differential equations, n_states = number of differential equations in the system.
+    n_odeparams : int
+        Number of parameters in the differential equation.
+
+    .. code-block:: python
+
+        def odefunc(y,t,p):
+            #Logistic differential equation
+            return p[0]*y[0]*(1-y[0])
+       
+        times = np.arange(0.5, 5, 0.5)
+
+        ode_model = DifferentialEquation(func = odefunc, t0 = 0, times = times, n_states = 1, n_odeparams = 1)
+    '''
+
+    __props__ = ()

```

Comment:
The props should not be empty. You can set it to `'func', 't0', 'times', 'n_states', 'n_odeparams'`.